# This is free and unencumbered software released into the public domain.

# This patches how the S3 configuration file is loaded so that the 'secretKey'
# values are used as paths from where the actual secrets are read from.
# https://github.com/seaweedfs/seaweedfs/wiki/s3-credentials

--- auth_credentials.go
+++ auth_credentials.go

@@ -339,9 +339,14 @@
 			t.Actions = append(t.Actions, Action(action))
 		}
 		for _, cred := range ident.Credentials {
+			secretKey, err := os.ReadFile(cred.SecretKey)
+			if err != nil {
+				glog.Warningf("unable to read secret key at %s", cred.SecretKey)
+				return err
+			}
 			t.Credentials = append(t.Credentials, &Credential{
 				AccessKey: cred.AccessKey,
-				SecretKey: cred.SecretKey,
+				SecretKey: string(secretKey),
 				Status:    cred.Status, // Load access key status
 			})
 			accessKeyIdent[cred.AccessKey] = t
 		}
